src/common/config_parser.cpp

```diff
@@ -679,16 +700,22 @@ void ConfigParser::addOptionsTranslation(cli::CLIWrapper& cli) {
       "Keep the output segmented into SentencePiece subwords");
 #endif
 
-  addSuboptionsInputLength(cli);
-  addSuboptionsTSV(cli);
-  addSuboptionsDevices(cli);
-  addSuboptionsBatching(cli);
+  // For self-adaptive translation these options are already added in
+  // `addOptionsTraining`
+  if(mode_ != cli::mode::selfadaptive) {
+    addSuboptionsInputLength(cli);
+    addSuboptionsTSV(cli);
```

Comment:
Does self-adaptive work with `--tsv`?

Reply:
It doesn't. TSV support would make sense for adaptive context reading, but it's implemented in `AdaptiveContextReader` separately from other data reading related code which has TSV support. I'll remove the TSV options.